runtime.go

```diff
@@ -512,42 +512,52 @@ func (r *Runtime) typeErrorResult(throw bool, args ...interface{}) {
 	}
 }
 
-func (r *Runtime) MemUsage(ctx *MemUsageContext) (uint64, error) {
-	total := uint64(0)
+func (r *Runtime) MemUsage(ctx *MemUsageContext) (memUsage uint64, newMemUsage uint64, err error) {
+	if r == nil {
+		return SizeEmpty, SizeEmpty, err
+	}
 
 	if r.globalObject != nil {
-		inc, err := r.globalObject.MemUsage(ctx)
-		total += inc
+		inc, newInc, err := r.globalObject.MemUsage(ctx)
+		memUsage += inc
+		newMemUsage += newInc
 		if err != nil {
-			return total, err
+			return memUsage, newMemUsage, err
 		}
 	}
 
-	for idx := range r.vm.callStack {
-		inc, err := r.vm.callStack[idx].MemUsage(ctx)
-		total += inc
-		if err != nil {
-			return total, err
+	if r.vm != nil {
```

Comment:
[opt] can do an early exit to remove the additional layer of conditional blocks.

```
if r.vm == nil{
  return memUsage, newMemUsage, nil
}
```